### DISCUSSION TITLE
Egress conenctions.

### DISCUSSION THREAD

**Discussion by ecarvalho-olisipo** [soluble-ai/kubetap#19] [Q&A]
Is this plugin able to intercept egress connections too? Tried here and could also see the incoming requests.

**Comment by gberche-orange** [comment]
AFAIK, egress connection from K8S containers don't use K8S services. Some service mesh products expose egress gateways instead.

Since kubetap focuses on K8S services, I understand that inspecting egress traffic from K8S pods is out of scope of kubetap.

**Reply by gberche-orange** [reply]
See related great promising work by @xxradar in https://xxradar.medium.com/mitmproxy-and-kubernetes-e897e903b1cb and associated repo:
 * https://github.com/xxradar/mitmproxy_k8s_interception/blob/main/docs/kubernetes_mitm_intercepts_basic.md
 * https://github.com/xxradar/mitmproxy_k8s_interception/blob/main/docs/kubernetes_mitm_intercepts_advanced.md
 * https://github.com/xxradar/mitmproxy_k8s_interception/blob/main/docs/hacking_ideas.md
